src/sentry/api/serializers/models/rule.py

```diff
@@ -43,19 +51,23 @@ def get_attrs(self, item_list, user, **kwargs):
 
     def serialize(self, obj, attrs, user):
         environment = attrs["environment"]
+        all_conditions = [
+            dict(list(o.items()) + [("name", _generate_rule_label(obj.project, obj, o))])
+            for o in obj.data.get("conditions", [])
+        ]
+
         d = {
             # XXX(dcramer): we currently serialize unsaved rule objects
             # as part of the rule editor
             "id": six.text_type(obj.id) if obj.id else None,
-            "conditions": [
-                dict(list(o.items()) + [("name", _generate_rule_label(obj.project, obj, o))])
-                for o in obj.data.get("conditions", [])
-            ],
+            "conditions": filter(lambda condition: not _is_filter(condition), all_conditions),
+            "filters": filter(lambda condition: _is_filter(condition), all_conditions),
```

Comment:
maybe commenting what you said here https://github.com/getsentry/sentry/pull/20412/files#r477411239 in some way, like filters aren't new conditions, they are the subset of conditions that pertain to event attributes (afaict)